Native installers

shannah opened the pull request:
This PR adds CI for generating native desktop installers for Mac, Windows, and Linux on each commit using [jDeploy](https://www.jdeploy.com).

*Why?*

- Since grokk is a desktop application, having to launch it via the command-line is awkward.  It is easier for users to be able to install the app into their Dock, Start Menu, etc.. so that it appears as a first class desktop application.

*Features*
- workflow runs on "tag" to automatically add native installers to each release.
- same workflow runs on each commit to automatically generate native installers for the latest version of each branch.
- Auto-update:  After user has installed the app, it will automatically check for updates on each launch, and update to the latest version.

*Example*
- You can download installers that are sync'd to my fork's "master" branch [here](https://github.com/shannah/brokk/releases/tag/master)

I haven't created any releases on my fork, but if I did, the release page would include links just like in the master branch example above.

Here's what the installer looks like on Mac.

<img width="670" height="545" alt="image" src="https://github.com/user-attachments/assets/84258e67-a100-48ce-bbcf-7feef71441df" />


Disclosure: I'm the creator of jDeploy.  I heard about brokk on the airhacks podcast - I remember listening about a year ago also when you were discussing JVector - I found your descriptions of how embeddings work to be very accessible - and that podcast motivated me to start tinkering with AI more.

If jDeploy isn't a fit for your project, that's okay.  I just thought I'd share "this cool" thing I did that makes brokk just a little bit better.

Great work on Brokk.  It's nice to see greenfield projects built with Swing in the wild.

jbellis commented:
Wow, thanks!  JDeploy was on my short list for native installers in the medium term, much appreciated!

Do you mind signing the attached CLA and sending to [my username]@brokk.ai?

Can also get you an e-signature version if you send me your email.
[Brokk CLA.pdf](https://github.com/user-attachments/files/21693039/Brokk.CLA.pdf)

shannah commented on package.json:
```diff
@@ -0,0 +1,33 @@
+        "yauzl": "^2.10.0",
+        "shelljs": "^0.8.4"
+    },
+    "license": "ISC",
```
@cmanto Yeah... that's the default boilerplate.  I can change to the correct one, whatever that is.

shannah commented:
I moved the jdeploy step into the release workflow.  Generated a sample release in my fork:
https://github.com/shannah/brokk/releases/tag/0.1.12

I also updated the license in the package.json to GPLv3.
For now, I have kept the jdeploy workflow, but only to generate for the master branch commits.  I looked at moving it into the ci workflow but, because that one is set to run on pull requests also, and it's intention is to "validate" the commits, I decided it wasn't a good fit to generate native bundles as part of that workflow.

shannah commented:
> It seems most of the time devs will launch directly from the command line or intellij, but I don't have a full sense. what you have now seems fine and we can adjust later if needed

How does launching from IntelliJ work?  I didn't see that option.
